### DISCUSSION TITLE
Automatic NGINX/Apache installation

### DISCUSSION THREAD

**Discussion by ghost** [searxng/searxng#2634] [Ideas]
I propose moving automatic installation of NGINX/Apache to the `Installation Script`, while keeping separate pages for the manual guide.

The reason is that while I was installing it, I missed one line and got confused about whether I need to complete instructions on the rest of the page(manual installation).

Also, I suggest including selection of HTTP server (or not installing any) to the installation script, instead of making it a separate article/instructions.

**Comment by return42** [comment]
> The reason is that while I was installing it, I missed one line and got confused about whether I need to complete instructions on the rest of the page(manual installation).

The documentation of the "Installation Script" procedure is here ..

https://docs.searxng.org/admin/installation-scripts.html

At the end there is a hint about HTTP proxy setup

![image](https://github.com/searxng/searxng/assets/554536/c970ab82-5799-4a30-9c7c-b356e4d8bece)

Not sure how we can improve it ..

> Also, I suggest including selection of HTTP server (or not installing any) to the installation script, instead of making it a separate article/instructions.

Installation of SearXNG services and HTTP server are different tasks and I think we should not mix up these both tasks which each other.. most maintainers already have a HTTP setup and might be confused if they have to decide to install a HTTP server while installing SearXNG services.
